drivers/counter/counter_native_posix.c

```diff
@@ -84,14 +99,38 @@ static int ctr_set_top_value(const struct device *dev,
 			     const struct counter_top_cfg *cfg)
 {
 	ARG_UNUSED(dev);
-	ARG_UNUSED(cfg);
 
-	posix_print_warning("%s not supported\n", __func__);
-	return -ENOTSUP;
+	if (is_alarm_pending) {
+		posix_print_warning("Can't set top value while alarm is active\n");
+		return -EBUSY;
+	}
+
+	if (cfg->flags & COUNTER_TOP_CFG_DONT_RESET) {
+		if (hw_counter_get_value() > cfg->ticks) {
+			if (cfg->flags & COUNTER_TOP_CFG_RESET_WHEN_LATE) {
+				hw_counter_reset();
+			}
+			return -ETIME;
+		}
+	} else {
+		hw_counter_reset();
+	}
+
+	top = *cfg;
+	is_top_set = true;
```

Comment:
I'm not sure how native_posix timer works but what if newly set value is TOP_VALUE without callback? Shouldn't it be reset to `false` and interrupt disabled?

Reply:
I initially didn't add that because it's not mentioned in Counter API docs and the `set_alarm` function has no such behavior. But, I agree it's a good idea (without it there's no way of setting an alarm after setting a top value without re-init), so I've added it in. Not sure if there is a good place to document that behavior or if it's already mentioned on some other page though.